defaults/main.yml

```diff
@@ -1,9 +1,69 @@
 ---
-# What packages to exclude in yum.conf at install time
-yum_exclude: ""
+#################################################
+# Common settings ###############################
+#################################################
 
 # Time to wait for install to finish (in minutes)
 install_timeout: "30"
 
+# Temp directory for storing provisioning related files
+provision_tempdir: "/tmp"
+
+# Where libvirt stores images and vms
+libvirt_root: "/var/lib/libvirt"
+
+# Root disk size (GB)
+root_disk_size: 30
+
+# Default to a single vcpu
+vcpus: 1
+
+# Default to 1GB of memory
+ram: 1024
+
+# Default environment name (suggested use case: dev/test/prod)
+environment: default-environment
+
+#################################################
+# Kickstart based deployment settings ###########
+#################################################
+
+# Disks
+disks:
+  - path: "{{ libvirt_root }}/{{ fqdn }}.raw"
+    size: "{{ root_disk_size }}"
+
+# What packages to exclude in yum.conf at install time
+yum_exclude: ""
+
 # Any additional packages to install during provisioning (as a list)
 extra_packages: []
+
+# Empty environment for running virt-install
+env_virt_install: {}
+
+#################################################
+# Image based deployment settings ###############
+#################################################
+
+# Default download location of CentOS image
+image_url_prefix: http://cloud.centos.org/centos/7/images/
+image_name: CentOS-7-x86_64-GenericCloud.qcow2
+
+# Where to download the images
+image_path: "{{ libvirt_root }}/images"
+timezone: Europe/Helsinki
+
+# DNS servers
+resolv_nameservers:
+  - 8.8.4.4
+  - 8.8.8.8
+
+#
+# If you want to add all your users, use an `adminusers` dictionary as required
```

Comment:
The `adminusers` and `default_gateway` variables are mandatory for image based deployments, but they are not defined in defaults. `adminusers` is mentioned as an optional parameter.